src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -1453,6 +1528,30 @@ public sealed partial class ProxySupportAnnotation : IResourceAnnotation
         public required bool ProxyEnabled { get { throw null; } set { } }
     }
 
+    public sealed partial class PublishingCallbackAnnotation : IResourceAnnotation
```

Comment:
We should mark these new PublishingX APIs experimental @davidfowl

Reply:
Yes I think so

Reply:
@mitchdenny are you doing this (and backporting)?

Reply:
I'm doing this, and all the remaining API review comments.